WordPress/Classes/Services/ReaderSearchSuggestionService.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+
+///
+///
+@objc class ReaderSearchSuggestionService : LocalCoreDataService
+{
+
+    ///
+    ///
+    func createOrUpdateSuggestionForPhrase(phrase: String) {
+        var suggestion = findSuggestionForPhrase(phrase)
+        if suggestion == nil {
+            suggestion = NSEntityDescription.insertNewObjectForEntityForName(ReaderSearchSuggestion.classNameWithoutNamespaces(),
+                                                                             inManagedObjectContext: managedObjectContext) as? ReaderSearchSuggestion
+            suggestion?.searchPhrase = phrase
+        }
+        suggestion?.date = NSDate()
+        ContextManager.sharedInstance().saveContext(managedObjectContext)
+    }
+
+
+    ///
+    ///
+    func findSuggestionForPhrase(phrase: String) -> ReaderSearchSuggestion? {
+        let fetchRequest = NSFetchRequest(entityName: "ReaderSearchSuggestion")
+        fetchRequest.predicate = NSPredicate(format: "searchPhrase MATCHES[cd] %@", phrase)
+
+        var suggestions = [ReaderSearchSuggestion]()
+        do {
+            suggestions = try managedObjectContext.executeFetchRequest(fetchRequest) as! [ReaderSearchSuggestion]
+        } catch let error as NSError {
+            DDLogSwift.logError("Error fetching search suggestion for phrase \(phrase) : \(error.localizedDescription)")
+        }
+
+        return suggestions.first
+    }
+
+
+    ///
+    ///
+    func fetchSuggestionsLikePhrase(phrase: String) -> [ReaderSearchSuggestion] {
+        let fetchRequest = NSFetchRequest(entityName: "ReaderSearchSuggestion")
+        fetchRequest.predicate = NSPredicate(format: "searchPhrase BEGINSWITH[cd] %@", phrase)
+
+        let sort = NSSortDescriptor(key: "date", ascending: false)
+        fetchRequest.sortDescriptors = [sort]
+
+        var suggestions = [ReaderSearchSuggestion]()
```

Comment:
True, but this way if the `try` block fails for any reason we still have a valid return value.